core-web/libs/data-access/src/lib/dot-workflow/dot-workflow.service.spec.ts

```diff
@@ -1,66 +1,50 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
+import { createHttpFactory, HttpMethod, SpectatorHttp } from '@ngneat/spectator/jest';
 
-/* eslint-disable @typescript-eslint/no-explicit-any */
-
-import { HttpTestingController, HttpClientTestingModule } from '@angular/common/http/testing';
-import { TestBed, getTestBed } from '@angular/core/testing';
-
-import { CoreWebService } from '@dotcms/dotcms-js';
-import { CoreWebServiceMock, mockWorkflows } from '@dotcms/utils-testing';
+import { mockWorkflows } from '@dotcms/utils-testing';
 
 import { DotWorkflowService } from './dot-workflow.service';
+import { DotCMSWorkflowMock, mockWorkflowstatus } from './utils/mocks';
 
 describe('DotWorkflowService', () => {
-    let injector: TestBed;
-    let dotWorkflowService: DotWorkflowService;
-    let httpMock: HttpTestingController;
+    let spectator: SpectatorHttp<DotWorkflowService>;
+    const createHttp = createHttpFactory(DotWorkflowService);
 
-    beforeEach(() => {
-        TestBed.configureTestingModule({
-            imports: [HttpClientTestingModule],
-            providers: [
-                { provide: CoreWebService, useClass: CoreWebServiceMock },
-                DotWorkflowService
-            ]
-        });
-        injector = getTestBed();
-        dotWorkflowService = injector.get(DotWorkflowService);
-        httpMock = injector.get(HttpTestingController);
-    });
+    beforeEach(() => (spectator = createHttp()));
 
     it('should get workflows', () => {
-        dotWorkflowService.get().subscribe((res: any) => {
-            expect(res).toEqual([
-                {
-                    hello: 'world',
-                    hola: 'mundo'
-                }
-            ]);
+        spectator.service.get().subscribe((res) => {
+            expect(res).toEqual(DotCMSWorkflowMock);
         });
 
-        const req = httpMock.expectOne('v1/workflow/schemes');
-        expect(req.request.method).toBe('GET');
-        req.flush({
-            entity: [
-                {
-                    hello: 'world',
-                    hola: 'mundo'
-                }
-            ]
+        spectator.expectOne('/api/v1/workflow/schemes', HttpMethod.GET).flush({
+            entity: DotCMSWorkflowMock
         });
     });
 
     it('should get default workflow', () => {
         const defaultSystemWorkflow = mockWorkflows.filter((workflow) => workflow.system);
 
-        dotWorkflowService.getSystem().subscribe((res: any) => {
+        spectator.service.getSystem().subscribe((res) => {
             expect(res).toEqual(defaultSystemWorkflow[0]);
```

Comment:
Probably need... always check if the test fail first ;)

Reply:
Added!

Reply:
First i added the done() and test fails, searching in spectator doc i dont see they uses done, i think the exceptOne() can check the result without the done function, so that is not needed
https://ngneat.github.io/spectator/docs/testing-with-http